internal/cmd/login.go

```diff
@@ -146,29 +146,44 @@ func login(options loginCmdOptions) error {
 				return err
 			}
 		}
-
 	}
 
 	return loginToInfra(client, loginReq)
 }
 
 func loginToInfra(client *api.Client, loginReq *api.LoginRequest) error {
 	loginRes, err := client.Login(loginReq)
-	if api.ErrorStatusCode(err) == http.StatusUnauthorized {
+	if err != nil {
 		switch {
 		case loginReq.AccessKey != "":
-			return &FailedLoginError{getLoggedInIdentityName(), accessKeyLogin}
+			return &LoginError{Message: "your access key may be invalid"}
 		case loginReq.PasswordCredentials != nil:
-			return &FailedLoginError{getLoggedInIdentityName(), localLogin}
+			return &LoginError{Message: "your username or password may be invalid"}
 		case loginReq.OIDC != nil:
-			return &FailedLoginError{getLoggedInIdentityName(), oidcLogin}
+			return &LoginError{
+				Message:    "please contact an administrator and check identity provider configurations",
+				ProviderID: loginReq.OIDC.ProviderID,
+			}
 		}
-	}
-	if err != nil {
+
 		return err
 	}
 
-	fmt.Fprintf(os.Stderr, "  Logged in as %s\n", termenv.String(loginRes.Name).Bold().String())
+	if loginRes.PasswordUpdateRequired {
```

Comment:
As mentioned in the PR description, the password update request needs to be handled before writing configs to disk. Otherwise the user can gets a session without completing the workflow

Reply:
Not for this PR, but I wonder if we should make changes to the API so that the one-time-password session only allows them to change their password. That way they can continue to attempt to update it without being locked out.

With this change, if they ctrl-c at this prompt, they'll be locked out until an admin resets their one-time-password, is that right?

Reply:
You're correct that the OTP will be expended as soon as they login even if they don't change their password. The behaviour you're asking about is captured in #1441 but supporting that behaviour requires substantial changes to the backend service so it's out of scope of this PR